roles/nuage_master/tasks/main.yaml

```diff
@@ -0,0 +1,35 @@
+---
+- name: Install Nuage Kubemon
+  sudo: true
+  yum: name={{ nuage_kubemon_rpm }} state=present
+
+- name: set kubeconfig 
+  sudo: true
+  lineinfile: dest={{ kubemon_yaml }} regexp=^kubeconfig line='kubeconfig{{':'}} {{ admin_config }}'
+
+- name: set openshift admin 
+  sudo: true
+  lineinfile: dest={{ kubemon_yaml }} regexp=^openshiftAdmin line='openshiftAdmin{{':'}} {{ os_admin }}'
+
+- name: Set Openshift master config path
+  sudo: true
+  lineinfile: dest={{ kubemon_yaml }} regexp=^openshiftMasterConfig line='openshiftMasterConfig{{':'}} {{ master_config_yaml }}'
```

Comment:
@abhat my apologies for not providing more context. I should have said the openshift_facts role. Within it, we provide an ansible module called openshift_facts that can be used for setting, persisting, and getting facts from the hosts. It would only be available during installation, reconfiguration (through ansible) and upgrade.